packages/kong-plugin/kong/plugins/readme-plugin/handler.lua

```diff
@@ -0,0 +1,244 @@
+  end
+
+  parsed_url = url.parse(host_url)
+  if not parsed_url.port then
+    if parsed_url.scheme == "http" then
+      parsed_url.port = 80
+
+    elseif parsed_url.scheme == "https" then
+      parsed_url.port = 443
+    end
+  end
+  if not parsed_url.path then
+    parsed_url.path = "/"
+  end
+
+  parsed_urls_cache[host_url] = parsed_url
+
+  return parsed_url
+end
+
+-- Creates a payload for ReadMe
+-- @return JSON string in ReadMe API Log format
+local function make_readme_payload(conf, entries)
+  local payload = {}
+  for _, entry in ipairs(entries) do
+    local request_entry = {
+      pageref = entry.request.url,
+      startedDateTime = os.date("!%Y-%m-%dT%H:%M:%SZ", entry.started_at / 1000),
+      time = entry.latencies.request,
+      request = {
+        httpVersion = entry.request.httpVersion,
+        headers = {},
+        method = entry.request.method,
+        queryString = {},
+        bodySize = entry.request.size,
+        url = entry.request.url,
+      },
+      response = {
+        status = entry.response.status,
+        headers = {},
+        content = {
+          mimeType = entry.response.mimeType,
+          size = entry.response.size,
+        },
+        bodySize = entry.response.size
+      }
+    }
+
+
+    -- Convert headers
+    for name, value in pairs(entry.request.headers) do
+      local final_value = value
+      if conf.hide_headers[name] then
+        if conf.hide_headers[name] == "" then
+          kong.log.debug("Hiding request header: ", name)
+          break
+        else
+          final_value = conf.hide_headers[name]
+          kong.log.debug("Overriding request header: ", name, " with ", final_value)
+        end
+      end
+      table.insert(request_entry.request.headers, {name = name, value = final_value})
+    end
+
+    for name, value in pairs(entry.response.headers) do
+      local final_value = value
+      if conf.hide_headers[name] then
+        if conf.hide_headers[name] == "" then
+          kong.log.debug("Hiding response header: ", name)
+          break
+        else
+          final_value = conf.hide_headers[name]
+          kong.log.debug("Overriding response header: ", name, " with ", final_value)
+        end
+      end
+      table.insert(request_entry.response.headers, {name = name, value = final_value})
+    end
+
+    for name, value in pairs(entry.request.querystring) do
+      table.insert(request_entry.request.queryString, {name = name, value = value})
+    end
+
+    local group = {}
+    if conf.group_by then
+      for header_name, group_value in pairs(conf.group_by) do
+        local header_value = entry.request.headers[header_name]
+        group[group_value] = header_value or "unknown"
+      end
+    end
+
+    local id_header = (conf.id_header and conf.id_header:lower()) or "authorization"
+    local id = entry.request.headers[id_header] or "unknown"
+    if id_header == "authorization" and entry.raw_auth then
+      id = entry.raw_auth
```

Comment:
We might want to mask this header via a `sha512` digest, similar to our other SDKs.

Reply:
do we mask it at this level or at the ingestion later? happy to do that here if we don't need the real one upstream

Reply:
updated to hash id